src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -763,6 +763,100 @@ bool IFloatingPoint<double>.TryWriteSignificandLittleEndian(Span<byte> destinati
         /// <inheritdoc cref="IFloatingPointIeee754{TSelf}.ILogB(TSelf)" />
         public static int ILogB(double x) => Math.ILogB(x);
 
+        /// <inheritdoc cref="IFloatingPointIeee754{TSelf}.MaxMagnitudeNumber(TSelf, TSelf)" />
+        public static double MaxMagnitudeNumber(double x, double y)
+        {
+            // This matches the IEEE 754:2019 `maximumMagnitudeNumber` function
+            //
+            // It does not propagate NaN inputs back to the caller and
+            // otherwise returns the input with a larger magnitude.
+            // It treats +0 as larger than -0 as per the specification.
+
+            double ax = Abs(x);
+            double ay = Abs(y);
+
+            if ((ax > ay) || IsNaN(ay))
```

Comment:
Two questions:
1. If ax was NaN and ay was 1.23, would the "(ax > ay)" condition return false?
2. If both ax and ay are NaN, ax would be returned here. Does that contradict with the comment "It does not propagate NaN inputs back to the caller"?

Reply:
1. Yes, `x cmp y` always returns `false` if either `x` or `y` is NaN
2. No, in the case that both are `NaN`, `NaN` is the thing to return